plugins/processors/ifname/ifname.go

```diff
@@ -139,15 +134,9 @@ func (d *IfName) Init() error {
 		return fmt.Errorf("parsing SNMP client config: %w", err)
 	}
 
-	if d.Translator == "gosmi" {
-		var err error
-		d.translator, err = si.NewGosmiTranslator(d.Path, d.Log)
-		if err != nil {
-			return err
-		}
-	} else {
-		d.translator = si.NewNetsnmpTranslator()
-	}
+	// Since OIDs in this plugin are always numeric there is no need
+	// to translate.
+	d.translator = si.NewNetsnmpTranslator()
```

Comment:
Then why initiating it anyway?

Reply:
Just because it needs to be something. I don't expect it to be used. We should consider changing ifname so it doesn't call table Init() at all. Then we won't need a translator at all.

Reply:
As said before, better would be the init not expect a translator as argument.

Reply:
What do you think about that @reimda?

Reply:
I don't see how we can make table.Init() not expect a translator. Here in ifname, Init() shouldn't need to look up anything in the mibs because we only use numeric oids, but in the main use case in the snmp input allows names so it does need to use mibs.

Maybe here in ifname we should change the code to manually initialize the table and then not call Init() at all. That would probably mean removing makeTableNoMock() completely and changing makeTable() to fill in all the fields of snmp.Table that table.Init() would.

Reply:
Easiest way would be by adding translator to the struct. But don't know if that is a good thing memory-wise.